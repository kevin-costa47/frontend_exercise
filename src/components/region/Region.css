.btn_show_forecast {
  display: inline-block;
  background: var(--light-color);
  color: #333;
  padding: 0.4rem 1.3rem;
  font-size: 1rem;
  border: none;
  cursor: pointer;
  margin-right: 0.5rem;
  transition: opacity 0.2s ease-in;
  outline: none;
  font-size: 0.8rem;
}

.regions_data_grid {
  display: grid;
  width: 100%;
  grid-template-columns: 1fr 100px;
  grid-gap: 1rem;
}
