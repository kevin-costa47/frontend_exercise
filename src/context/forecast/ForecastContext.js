import { createContext } from "react";

const forecastContext = createContext({
    forecasts: [],
    region: {},
    regions: [],
});

export default forecastContext;
