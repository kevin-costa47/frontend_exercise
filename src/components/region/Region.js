import React, { useContext } from "react";
import ForecastContext from "../../context/forecast/ForecastContext";
import "./Region.css";

const Region = () => {
  const forecastContext = useContext(ForecastContext);
  const { regions = [] } = forecastContext;

  return regions.map((region) => {
    const { globalIdLocal = "", idAreaAviso } = region;

    const idArea = `${idAreaAviso} ${globalIdLocal}`;

    return (
      <div className="card" key={globalIdLocal}>
        <div className="regions_data_grid">
          <h3 className="item-center">{idArea}</h3>
          <button
            className="btn_show_forecast"
            onClick={() => forecastContext.getForecast(region)}
          >
            Show
          </button>
        </div>
      </div>
    );
  });
};

Region.propTypes = {};

export default Region;
