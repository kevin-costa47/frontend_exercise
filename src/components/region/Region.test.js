import React from "react";
import { shallow, mount } from "enzyme";
import Region from "./Region";
import ForecastContext from "../../context/forecast/ForecastContext";

const renderWithContext = (children, value) => {
  return (
    <ForecastContext.Provider value={value}>
      {children}
    </ForecastContext.Provider>
  );
};

describe("Region", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const regions = [
    {
      idAreaAviso: "AVR",
      globalIdLocal: "1010500",
    },
  ];

  const getForecast = jest.fn();

  const context = {
    regions,
    getForecast,
  };

  it("Should render the Region Component correctly", () => {
    let wrapped = mount(renderWithContext(<Region />, context));

    expect(wrapped).toMatchSnapshot();
  });

  it("Simulates getForecast Event", () => {
    const button = mount(renderWithContext(<Region />, context));

    button.find("button").simulate("click");
    expect(getForecast.mock.calls.length).toEqual(1);
  });
});
