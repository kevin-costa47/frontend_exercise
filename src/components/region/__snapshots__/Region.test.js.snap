// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Region should render the Region Component correctly 1`] = `
<Region>
  <div
    className="data_row"
    key="1010500"
  >
    <div
      className="grid_regions_data"
    >
      <h3
        className="all-center"
      >
        AVR 1010500
      </h3>
      <button
        className="btn_show_forecast"
        onClick={[Function]}
      >
        Show
      </button>
    </div>
  </div>
</Region>
`;
